.circleci/config.yml

```diff
@@ -59,26 +72,19 @@ jobs:
     steps:
       - restore_cache:
           key: v1-specs-repo-{{ .Branch }}-{{ .Revision }}
-      - restore_cached_venv:
-          venv_name: v1-pyspec-03
-          reqs_checksum: '{{ checksum "test_libs/pyspec/requirements.txt" }}-{{ checksum "test_libs/pyspec/requirements-testing.txt" }}'
+      - restore_default_cached_venv
```

Comment:
1. Let's not mingle caches of different venvs, there's no "default"
2. Let's reduce the checksum complexity to 1 hash, and follow the standard
3. Let's don't use double version tickers in the caching key
4. Multiple parts split by a `-` are likely the root of the cache-key problems. See point 2.

Caching is a nightmare sometimes, but we can get this right :)
Marking this PR is blocked for now, caching is not quite ready to merge currently.